### PR TITLE
couch -> sql migration changes

### PR DESCRIPTION
**Pull request by snopoke**
Best read by commit :fish:

### CODE REVIEW

File: corehq/apps/couch_sql_migration/couchsqlmigration.py [hunk 1]
```diff
@@ -207,13 +213,30 @@ def _diff_cases(self, couch_cases):
             couch_case = couch_cases[sql_case.case_id]
             sql_case_json = sql_case.to_json()
             diffs = json_diff(couch_case, sql_case_json, track_list_indices=False)
-            self.diff_db.add_diffs(
-                couch_case['doc_type'], sql_case.case_id,
-                filter_case_diffs(couch_case, sql_case_json, diffs, self.forms_that_touch_cases_without_actions)
-            )
+            diffs = filter_case_diffs(couch_case, sql_case_json, diffs, self.forms_that_touch_cases_without_actions)
```
Comment (dimagimon) [inline corehq/apps/couch_sql_migration/couchsqlmigration.py]: E501 line too long (116 > 115 characters)

File: corehq/apps/couch_sql_migration/couchsqlmigration.py [hunk 2]
```diff
@@ -207,13 +213,30 @@ def _diff_cases(self, couch_cases):
...
+    def _rebuild_couch_case_and_re_diff(self, couch_case, sql_case_json):
+        from corehq.form_processor.backends.couch.processor import FormProcessorCouch
+        from corehq.apps.tzmigration.timezonemigration import json_diff
+
+        rebuilt_case = FormProcessorCouch.hard_rebuild_case(
+            self.domain, couch_case['_id'], None, save=False
+        )
+        rebuilt_case_json = rebuilt_case.to_json()
+        diffs = json_diff(rebuilt_case_json, sql_case_json, track_list_indices=False)
+        diffs = filter_case_diffs(rebuilt_case_json, sql_case_json, diffs, self.forms_that_touch_cases_without_actions)
```
Comment (dimagimon) [inline corehq/apps/couch_sql_migration/couchsqlmigration.py]: E501 line too long (119 > 115 characters)

File: corehq/apps/couch_sql_migration/management/commands/migrate_domain_from_couch_to_sql.py [hunk 3]
```diff
@@ -94,47 +97,48 @@ def print_stats(self, domain, short=True, diffs_only=False):
...
 
         form_ids_in_couch = set(get_doc_ids_in_domain_by_type(
             domain, "XFormInstance-Deleted", XFormInstance.get_db())
         )
         form_ids_in_sql = set(FormAccessorSQL.get_deleted_form_ids_in_domain(domain))
-        diff_count = diff_stats.pop("XFormInstance-Deleted", 0)
+        diff_count, num_docs_with_diffs = diff_stats.pop("XFormInstance-Deleted", (0, 0))
         has_diffs |= self._print_status(
-            "XFormInstance-Deleted", form_ids_in_couch, form_ids_in_sql, diff_count, short, diffs_only
+            "XFormInstance-Deleted", form_ids_in_couch, form_ids_in_sql, diff_count, num_docs_with_diffs, short, diffs_only
```
Comment (dimagimon) [inline corehq/apps/couch_sql_migration/management/commands/migrate_domain_from_couch_to_sql.py]: E501 line too long (123 > 115 characters)

File: corehq/apps/couch_sql_migration/management/commands/migrate_domain_from_couch_to_sql.py [hunk 4]
```diff
@@ -94,47 +97,48 @@ def print_stats(self, domain, short=True, diffs_only=False):
...
 
         case_ids_in_couch = set(get_doc_ids_in_domain_by_type(
             domain, "CommCareCase-Deleted", XFormInstance.get_db())
         )
         case_ids_in_sql = set(CaseAccessorSQL.get_deleted_case_ids_in_domain(domain))
-        diff_count = diff_stats.pop("CommCareCase-Deleted", 0)
+        diff_count, num_docs_with_diffs = diff_stats.pop("CommCareCase-Deleted", (0, 0))
         has_diffs |= self._print_status(
-            'CommCareCase-Deleted', case_ids_in_couch, case_ids_in_sql, diff_count, short, diffs_only
+            'CommCareCase-Deleted', case_ids_in_couch, case_ids_in_sql, diff_count, num_docs_with_diffs, short, diffs_only
```
Comment (dimagimon) [inline corehq/apps/couch_sql_migration/management/commands/migrate_domain_from_couch_to_sql.py]: E501 line too long (122 > 115 characters)

**Review by benrudolph — APPROVED** [review]
looks good and makes sense to me. couple of lint errors

**Pull request merged**